pkg/apis/application/v1alpha1/types.go

```diff
@@ -1277,7 +1287,8 @@ func (p *AppProject) GetRoleByName(name string) (*ProjectRole, int, error) {
 }
 
 // GetJWTToken looks up the index of a JWTToken in a project by id (new token), if not then by the issue at time (old token)
-func (p *AppProject) GetJWTToken(roleName string, issuedAt int64, id string) (*JWTToken, int, error) {
+func (p *AppProject) GetJWTTokenFromSpec(roleName string, issuedAt int64, id string) (*JWTToken, int, error) {
```

Comment:
I think we should not use a deprecated field internally. Instead perform project normalization and then use `GetJWTToken` method.

Reply:
When deleting a JWTToken, currently we delete in both spec and status. This method GetJWTTokenFromSpec is support delete a token from spec.

Reply:
Thank you for the explanation @mayzhang2000 ! 

I want to make sure we have as least a possible code that uses an obsolete field. Can you please add `AppProject.RemoveToken(roleName string, issuedAt int64, id string)` method encapsulates removing logic. Internally that method can check tokens in both status and spec.